src/Compilers/Server/VBCSCompiler/NamedPipeClientConnection.cs

```diff
@@ -229,6 +202,44 @@ internal NamedPipeClientConnection(ICompilerServerHost compilerServerHost, strin
             }
         }
 
+        protected override void ValidateBuildRequest(BuildRequest request)
+        {
+            // Now that we've read data from the stream we can validate the identity.
+            if (!ClientAndOurIdentitiesMatch(_pipeStream))
+            {
+                throw new Exception("Client identity does not match server identity.");
+            }
+        }
+
+        /// <summary>
+        /// Does the client of "pipeStream" have the same identity and elevation as we do?
+        /// </summary>
+        private static bool ClientAndOurIdentitiesMatch(NamedPipeServerStream pipeStream)
```

Comment:
Were this and the function below simply moved or were they also changed in some way?